src/backend/distributed/metadata/node_metadata.c

```diff
@@ -757,6 +757,18 @@ master_update_node(PG_FUNCTION_ARGS)
 		if (force)
 		{
 			handle = StartLockAcquireHelperBackgroundWorker(MyProcPid, lock_cooldown);
+			if (!handle)
+			{
+				/*
+				 * set lock_timeout to lock_cooldown, because if there is no slot for
+				 * background worker, we don't want it to wait forever to get lock.
```

Comment:
Might be worth elaborating on why we're doing this:

```c
if (!handle)
{
    /*
     * We failed to start a background worker, which probably means that we exceeded 
     * max_worker_processes, and this is unlikely to be resolved by retrying. We do not want
     * to repeatedly throw an error because if master_update_node is called to complete a
     * failover then finishing is the only way to bring the cluster back up. Therefore we
     * give up on killing other backends and simply wait for the lock. We do set lock_timeout
     * to lock_cooldown, because we don't want to wait forever to get a lock.
     */
```